Spatial_Engine/Modify/Orient.cs

```diff
@@ -0,0 +1,79 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Base.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using BH.oM.Dimensional;
+using BH.oM.Geometry;
+using BH.oM.Geometry.CoordinateSystem;
+
+
+namespace BH.Engine.Spatial
+{
+    public static partial class Modify
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Orients the element from one coordinate system to another.")]
+        [Input("element", "The element to orient.")]
+        [Input("from", "The cartesian coordinate system to orient from. Deaults to global XY if nothing is provided.")]
+        [Input("to", "The cartesian coordinate system to orient to. Deaults to global XY if nothing is provided.")]
+        [Output("element", "The reoriented element.")]
+        public static IElement Orient(this IElement element, Cartesian from = null, Cartesian to = null)
```

Comment:
Personally I would consider adding a static `Cartesian.Global` or similar and defaulting `from` and `to` to that value. Now this method creates the risk of `null` being passed unnoticed, which is a no go for me I think.

Reply:
there is not static one (otherwise I would have used it), can ofc switch to create one with standard axis though.

Also thought about the potential issue if the default values changed, but I have included all three null providing cases for coordinate system in the UT, so if it was ever changed, the UT would flag it.

If you still think that is to risky, can ofc update?

Reply:
I would create one, same as we have e.g. in `Basis.XY`. Not sure about the UT part, but I am thinking about a scenario where another method crashes and returns `null` and then it is picked by `Orient` and processed as if the input was valid - this is what I am afraid of.

Reply:
Ah, I see. Not sure that is solvable from here though. Think the optional inputs need to be compile time constants, but will have a look.

Reply:
Ah, you are right, the coordinate system will not be constant compile time. In this case I am taking back my original suggestion - would rather be on the safe side than on the handy one.